auth/auth.go

```diff
@@ -408,18 +425,24 @@ func (c *Client) verifySessionCookie(ctx context.Context, sessionCookie string,
 
 // IsSessionCookieRevoked checks if the given error was due to a revoked session cookie.
 //
-// When IsSessionCookieRevoked returns true, IsSessionCookieInvalid is guranteed to return true.
+// When IsSessionCookieRevoked returns true, IsSessionCookieInvalid is guaranteed to return true.
 func IsSessionCookieRevoked(err error) bool {
 	return hasAuthErrorCode(err, sessionCookieRevoked)
 }
 
-func (c *baseClient) checkRevoked(ctx context.Context, token *Token) (bool, error) {
+// checkRevokedOrDisabled checks whether the input token has been revoked or disabled.
+func (c *baseClient) checkRevokedOrDisabled(ctx context.Context, token *Token) (revoked bool, disabled bool, err error) {
```

Comment:
Instead of expanding the return signature, it's probably better to just return a single error value from this function.

```
checkRevokedOrDisabled(ctx context.Context, token *Token, revokedErrCode string) err error
```

Then the caller can be just:

```go
if c.isEmulator || checkRevokedOrDisabled {
		if err := c.checkRevokedOrDisabled(ctx, decoded, idTokenRevoked); err != nil {
			return nil, err
		}
}
```